collection_size.go

```diff
@@ -110,20 +110,38 @@ func (u *CollectionSizeCollector) Describe(c chan<- *prometheus.Desc) {
 
 // Collect implements the prometheus.Collector interface.
 func (u *CollectionSizeCollector) Collect(ch chan<- prometheus.Metric) {
+	fail := func() {
+		mSize, err := prometheus.NewConstMetric(u.sizeDesc, prometheus.GaugeValue, float64(0))
```

Comment:
Is this the right thing to do? After all, if something has failed we'll try to report it as 0 which could be misleading, isn't it better to report nothing? What's the approach Prometheus expects us to use?